### PR TITLE
Editing - Run APM Server on ECK

### PR DESCRIPTION
**Pull request by alaudazzi**
This PR:

- reworks a few sentences
- fixes typos and capitalization
- adds links to facilitate the navigation

Relates to [#65636](https://github.com/elastic/cloud/issues/65636)

### CODE REVIEW

File: docs/orchestrating-elastic-stack-applications/apm-server.asciidoc [hunk 1]
```diff
@@ -75,9 +76,17 @@ apm-server-quickstart-apm-server-69b447ddc5-fflc6   1/1     Running   0
 [id="{p}-apm-advanced-configuration"]
 == Advanced configuration
 
+This section covers the following topics:
+
+** <<{p}-apm-agent-central-configuration,Use APM Agent central configuration>>
+** <<{p}-apm-customize-configuration,Customize the APM Server configuration>>
+** <<{p}-apm-secure-settings,APM Secrets keystore for secure settings>>
+** <<{p}-apm-existing-es,Reference an existing Elasticsearch cluster>>
```
Comment (charith-elastic) [inline docs/orchestrating-elastic-stack-applications/apm-server.asciidoc:84]: The overridden link text in some of the links don't match the actual title of the section they are linking to. Removing the explicit link text would keep them in sync.

```suggestion
** <<{p}-apm-agent-central-configuration>>
** <<{p}-apm-customize-configuration>>
** <<{p}-apm-secure-settings>>
** <<{p}-apm-existing-es>>
```

File: docs/orchestrating-elastic-stack-applications/apm-server.asciidoc [hunk 2]
```diff
@@ -155,25 +164,27 @@ spec:
...
-First, you need to store the certificate authority of the Elasticsearch cluster in a secret:
+You need to store the certificate authority of the Elasticsearch cluster in a secret:
 +
 [source,yaml]
 ----
 kubectl create secret generic es-ca --from-file=tls.crt=elasticsearch-ca.crt
 ----
 +
-NOTE: the file `elasticsearch-ca.crt` must contain the CA certificate of the Elasticsearch cluster you want to use with the APM Server.
+NOTE: The file `elasticsearch-ca.crt` must contain the CA certificate of the Elasticsearch cluster you want to use with the APM Server.
```
Comment (charith-elastic) [inline docs/orchestrating-elastic-stack-applications/apm-server.asciidoc:183]:
```suggestion
NOTE: The `elasticsearch-ca.crt` file must contain the CA certificate of the Elasticsearch cluster you want to use with the APM Server.
```

File: docs/orchestrating-elastic-stack-applications/apm-server.asciidoc [hunk 3]
```diff
@@ -223,6 +234,12 @@ For more details on how to configure the APM agents to work with custom certific
 [id="{p}-apm-connecting"]
 == Connect to the APM Server
 
+This section covers the following topics:
+
+** <<{p}-apm-service,APM Server service>>
+** <<{p}-apm-secret-token,APM Server secret token>>
```
Comment (charith-elastic) [inline docs/orchestrating-elastic-stack-applications/apm-server.asciidoc:240]:
```suggestion
** <<{p}-apm-service>>
** <<{p}-apm-secret-token>>
```

File: docs/orchestrating-elastic-stack-applications/apm-server.asciidoc (hunk 1 above)
Comment (alaudazzi) [reply docs/orchestrating-elastic-stack-applications/apm-server.asciidoc:84]: Nice! Thanks @charith-elastic

**Pull request merged**